src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -34,6 +35,64 @@ public CommandResult executeUndoableCommand() throws CommandException {
 
         List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
 
+        ArrayList<Integer> executableIdx = new ArrayList<>();
+        ArrayList<Integer> unExeIdx = new ArrayList<>();
+        Boolean hasExecutableIdx = false;
+
+        for (Index idx : targetIdxs) {
+            int intIdx = idx.getZeroBased();
+            if (intIdx < lastShownList.size()) {
+                executableIdx.add(intIdx);
+                hasExecutableIdx = true;
+            } else {
+                unExeIdx.add(intIdx);
+            }
+        }
+
+        if (!hasExecutableIdx) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_INDEX_ALL);
+        }
+
+
+        ArrayList<ReadOnlyPerson> toDeletePerson = new ArrayList<>();
+
+
+        for (int idx : executableIdx) {
+            try {
+                toDeletePerson.add(lastShownList.get(idx));
+                model.deletePerson(lastShownList.get(idx));
+            } catch (PersonNotFoundException e) {
+                assert false : "The target person cannot be missing";
+            }
+        }
+
+        StringBuilder sb = new StringBuilder();
```

Comment:
can create a new string builder method with @params toDeletePerson so you can use it for the test without writing the code again